core/src/lib/edit.cc

```diff
@@ -625,6 +625,29 @@ char* add_commas(char* val, char* buf)
   return buf;
 }
 
+
+// Add spaces to a string, which is presumably a number.
+char* add_spaces(char* val, char* buf)
+{
+  int len, nc;
+  char *p, *q;
+  int i;
+
+  if (val != buf) { strcpy(buf, val); }
+  len = strlen(buf);
+  if (len < 1) { len = 1; }
+  nc = (len - 1) / 3;
+  p = buf + len;
+  q = p + nc;
+  *q-- = *p--;
+  for (; nc; nc--) {
+    for (i = 0; i < 3; i++) { *q-- = *p--; }
+    *q-- = ' ';
+  }
+
+  return buf;
+}
+
```

Comment:
Is duplicating the code the best way to do this or do you think there is a better way to do solve this?
Do you see any room for improvement in the function declaration itself (i.e. parameter and return types)?

Reply:
> Is duplicating the code the best way to do this or do you think there is a better way to do solve this? Do you see any room for improvement in the function declaration itself (i.e. parameter and return types)?

yes I can do that, and that's much better I think, there is a slight issue since I'll need to maybe change the function calls in everyplace for add_commas or i can make it so default function call formats with commas.

Reply:
> Is duplicating the code the best way to do this or do you think there is a better way to do solve this? Do you see any room for improvement in the function declaration itself (i.e. parameter and return types)?

thankfully there weren't a lot of function calls to 'add_commas'  so I changed the function to add_separator